packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -84,21 +94,14 @@ export class RemoteMessageProcessor {
 	}
 }
 
-const copy = (remoteMessage: ISequencedDocumentMessage): ISequencedDocumentMessage => {
-	// Do shallow copy of message, as the processing flow will modify it.
-	// There might be multiple container instances receiving same message
-	// We do not need to make deep copy, as each layer will just replace message.content itself,
-	// but would not modify contents details
-	const message = { ...remoteMessage };
-
+/** Takes an incoming message and if the contents is a string, JSON.parse's it in place */
+const ensureContentsDeserialized = (mutableMessage: ISequencedDocumentMessage): void => {
 	// back-compat: ADO #1385: eventually should become unconditional, but only for runtime messages!
```

Comment:
However this ends up, is there a reason to use `const` definition here rather than a [named] `function`?

Reply:
I always wonder that too. Here's what Bing has to say on the matter:

## From Bing

In JavaScript, both `const` with a function type and a named function are used to define functions, but there are some differences between them:

1. **Hoisting**: Named functions are hoisted, which means they can be called before they are defined. On the other hand, `const` functions are not hoisted, so they must be declared before they are called.

2. **Reassignment**: `const` prevents reassignment of the function name while `function` does not.

3. **Lexical Context**: Arrow functions (which are often used with `const`) do not have their own lexical context, so they won't have a scoped `this` and can't be used as a constructor.

Here's an example of each:

```javascript
// Named function
function square(number) {
  return number * number;
}

// Const function
const square = (number) => {
  return number * number;
}
```
Both of these functions will return the square of a number.

Reply:
Mmm hmm. But is there a reason here?
In my experience an additional difference is that `function` has a name which is visible while debugging.

Reply:
#RealIntelligence

No I doubt there's a reason.  Not going to change it unless I have to take another change though, so I can go ahead and merge it sooner (CI takes a long time)

Reply:
If there's no use of `this` in the body of the function, I always make it a function. Advantage 1 above is occasionally useful, and I've never encountered a problem involving Disadvantage 2 above in typescript, because you can't rebind function symbols in typescript AFAIK.

Reply:
Certainly fine with not changing it. I mostly want guidance for code I write and giving feedback. I think I'll go with my common of `function` and call it out gently when I see otherwise. Thanks.